src/main/java/io/codurance/pouch/Resource.java

```diff
@@ -1,12 +1,16 @@
 package io.codurance.pouch;
 
 import org.springframework.data.annotation.Id;
+import org.springframework.data.annotation.Transient;
 import org.springframework.data.domain.Persistable;
 
 import java.time.Instant;
 import java.util.UUID;
 
-public class Resource implements Persistable {
+public class Resource implements Persistable<UUID> {
+
+    @Transient
+    private transient boolean persisted;
 
```

Comment:
Just to give my reviewers more context here: In **Spring Data JDBC** there is no dedicated repository method for updating an entity. It seems the developer needs to add logic for the program to decide whether or not a just instantiated entity is truly new or not. I already had to consider this when creating new resources. The case of an update required more work on the implementation of the `Persistable<ID>` interface.

See [Spring Data JDBC documentation on "Entity State Detection Strategies"](https://docs.spring.io/spring-data/jdbc/docs/1.1.0.M1/reference/html/#jdbc.entity-persistence.state-detection-strategies).

Reply:
This does seem very odd to me. Are we sure that we are going in the right direction here? My thought was to simply change the PK to be a UUID instead of a String. I wasn't aware that we wanted to start maintaining entity state. Perhaps we can do a call later to chat about all this.

Reply:
Hi there and thanks for the review. I think you may have gotten confused about the context of this PR. This is no longer about introducing `UUID` only. This is to introduce update functionality for my domain entity. However, I think I got your general concern here and got into a discussion with @alfredodev , resulting in a Blitz-Pairing to resolve this issue. In the refined solution there is less framework code and no more implementation of the `Persistable` interface. The issue with the "UPSERT" is avoided by using the ResourceDTO (and re-setting its values) to be handed down.

Reply:
Can I resolve this conversation? Are the concerns addressed sufficiently (see changes brought about in commits below)? The entity state is no longer part of this solution. This PR should be ready to merge into master.